cohere/client.py

```diff
@@ -796,3 +800,133 @@ def wait_for_bulk_embed_job(
+        """Get a finetune by id.
+
+        Args:
+            finetune_id (str): finetune id
+        Returns:
+            Finetune: the finetune
+        """
+        json = {"finetuneID": finetune_id}
+        response = self._client._request(f"{cohere.BLOBHEART_URL}/GetFinetune", method="POST", json=json)
+        return Finetune.from_dict(response["finetune"])
+
+    def get_by_name(self, name: str) -> Finetune:
+        """Get a finetune by name.
+
+        Args:
+            name (str): finetune name
+        Returns:
+            Finetune: the finetune
+        """
+        json = {"name": name}
+        response = self._client._request(f"{cohere.BLOBHEART_URL}/GetFinetuneByName", method="POST", json=json)
+        return Finetune.from_dict(response["finetune"])
+
+    def list(
+        self,
+        statuses: Optional[List[FINETUNE_STATUS]] = None,
+        before: Optional[datetime] = None,
+        after: Optional[datetime] = None,
+        limit: int = 50,
+        order_by: Optional[Literal["asc", "desc"]] = None,
+    ) -> List[Finetune]:
+        """List finetunes of your organization.
+
+        Args:
+            statuses (FINETUNE_STATUS, optional): search for fintunes which are in one of these states
+            before (datetime, optional): search for finetunes that were created before this timestamp
+            after (datetime, optional): search for finetunes that were created after this timestamp
+            limit (int, optional): maximum number of results
+            order_by (Literal["asc", "desc"], optional): sort finetunes by created at, either asc or desc
+        Returns:
+            List[Finetune]: a list of finetunes.
+        """
+        if before:
+            before = before.replace(tzinfo=before.tzinfo or timezone.utc)
+        if after:
+            after = after.replace(tzinfo=after.tzinfo or timezone.utc)
+
+        json = {
+            "query": {
+                "statuses": statuses,
+                "before": before.isoformat(timespec="seconds") if before else None,
+                "after": after.isoformat(timespec="seconds") if after else None,
+                "limit": limit,
+                "orderBy": order_by,
+            }
+        }
+
+        response = self._client._request(f"{cohere.BLOBHEART_URL}/ListFinetunes", method="POST", json=json)
+        return [Finetune.from_dict(r) for r in response["finetunes"]]
+
+    def create(self, name: str, finetune_type: FINETUNE_BASE_TYPE, dataset: Dataset) -> str:
+        """Create a new finetune
+
+        Args:
+            name (str): name of your finetune, has to be unique across your organization
+            finetune_type (GENERATIVE, CONTRASTIVE, CLASSIFICATION): type of finetune
+            dataset (InMemoryDataset, CsvDataset, JsonlDataset, TextDataset): A dataset for your training. Consists of a train and optional eval file.
+        Returns:
+            str: the id of the finetune that was created
```

Comment:
Can we return the `Finetune` object instead of the id?

Reply:
Usually yes, but the backend return a malformed object currently. Have to fix this first, then we can return the finetune object.